### PR TITLE
FIX - php V8 lib files

### PR DESCRIPTION
**Pull request by BB2A-Anthony**
# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "FIX", "CLOSE" or "NEW" section* (use uppercase to have the PR appears into the ChangeLog, lowercase will not appears)
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed texts with meaningful information*

# FIX|Fix #[*issue_number Short description*]
[*Long description*]

# CLOSE|Close #[*issue_number Short description*]
[*Long description*]

# NEW|New [*Short description*]
[*Long description*]

### CODE REVIEW

File: htdocs/core/lib/files.lib.php [hunk 1]
```diff
@@ -1674,11 +1674,12 @@ function dol_add_file_process($upload_dir, $allowoverwrite = 0, $donotupdatesess
 				}
 
 				// dol_sanitizeFileName the file name and lowercase extension
+				$infoTFile = pathinfo($upload_dir."/".$TFile['name'][$i]);
 				$info = pathinfo($destfull);
-				$destfull = $info['dirname'].'/'.dol_sanitizeFileName($info['filename'].($info['extension'] != '' ? ('.'.strtolower($info['extension'])) : ''));
+				$destfull = $info['dirname'].'/'.dol_sanitizeFileName($info['filename'].(!empty($info['extension']) ? ('.'.strtolower($info['extension'])) : (!empty($infoTFile['extension']) ? '.'.strtolower($infoTFile['extension']) : '')));
```
Comment (eldy) [inline htdocs/core/lib/files.lib.php:1679]: Why adding a backfall on infoTFile['extension'] ?
Do you have an exemple of value we have for file we upload and the result we get on $destfull for this case that need the $infoTFile ?

**Comment by bb2a** [comment]
@eldy it's not good?

File: htdocs/core/lib/files.lib.php [hunk 2]
```diff
@@ -1674,11 +1674,12 @@ function dol_add_file_process($upload_dir, $allowoverwrite = 0, $donotupdatesess
...
 				$info = pathinfo($destfile);
 
-				$destfile = dol_sanitizeFileName($info['filename'].($info['extension'] != '' ? ('.'.strtolower($info['extension'])) : ''));
+				$destfile = dol_sanitizeFileName($info['filename'].(!empty($info['extension']) ? ('.'.strtolower($info['extension'])) : (!empty($infoTFile['extension']) ? '.'.strtolower($infoTFile['extension']) : '')));
```
Comment (eldy) [inline htdocs/core/lib/files.lib.php:1682]: Same question than previous point here.

**Comment by BB2A-Anthony** [comment]
@eldy it's not good?

**Pull request closed without merging**